### PR TITLE
Vulkan: WIP: KHR_sampler_ycbc_conversion extension support: step 1

### PR DESCRIPTION
**Pull request by Qining**
Work in progress.

Ignore the old dependency graph in this CL as the old dependency graph
will be dropped soon.

Multi-planar format images are not fully supported yet. This CL is to
insert a 'plane' layer in the original image->aspect->layer->level
hierarchy, and move BoundMemory info and MemoryRequirements from image
object to plane object.

### CODE REVIEW

File: gapis/api/vulkan/BUILD.bazel [hunk 1]
```diff
@@ -63,8 +63,8 @@ go_library(
         "draw_call_mesh.go",
         "externs.go",
         "find_issues.go",
-        "footprint_builder.go",
         "graph_visualization.go",
+        # "footprint_builder.go",
```
Comment (AWoloszyn) [inline gapis/api/vulkan/BUILD.bazel]: Can we actually just remove these from the build, rather htan commenting them out.

File: gapis/api/vulkan/BUILD.bazel [hunk 2]
```diff
@@ -121,7 +121,7 @@ go_library(
         "//gapis/replay/protocol:go_default_library",
         "//gapis/replay/value:go_default_library",
         "//gapis/resolve:go_default_library",
-        "//gapis/resolve/dependencygraph:go_default_library",
+        # "//gapis/resolve/dependencygraph:go_default_library",
```
Comment (AWoloszyn) [inline gapis/api/vulkan/BUILD.bazel]: Same here

File: gapis/api/vulkan/BUILD.bazel [hunk 3]
```diff
@@ -153,7 +153,7 @@ go_test(
     name = "go_default_test",
     srcs = [
         "externs_test.go",
-        "footprint_builder_test.go",
+        # "footprint_builder_test.go",
```
Comment (AWoloszyn) [inline gapis/api/vulkan/BUILD.bazel]: Same

File: gapis/api/vulkan/api/synchronization.api [hunk 4]
```diff
@@ -625,11 +625,13 @@ sub void processMemoryBarriers(VkPipelineStageFlags       srcStageMask,
                                map!(u32, VkMemoryBarrier) memoryBarriers) {
   if len(memoryBarriers) > 0 {
     for _ , _ , img in Images {
-      for _ , _ , aspect in img.Aspects {
-        for _ , _ , layer in aspect.Layers {
-          for _ , _ , level in layer.Levels {
-            read(level.Data)
-            write(level.Data)
+      for _ , _ , plane in img.Planes {
```
Comment (AWoloszyn) [inline gapis/api/vulkan/api/synchronization.api]: I think we need to do something about this 5-level nested loop.

Comment (AWoloszyn) [reply gapis/api/vulkan/api/synchronization.api]: Which will turn into a 6-level nested loop with mutli-device stuff

File: gapis/api/vulkan/BUILD.bazel (hunk 1 above)
Comment (Qining) [reply gapis/api/vulkan/BUILD.bazel]: I will pull the stale code dropping part out and make another CL, this one will just be rebased on the new one.

**Comment by Qining** [comment]
This CL is going to be closed. The content here will be split into multiple smaller CLs.

**Pull request closed without merging**